### PR TITLE
Guided Workouts

### PR DESCRIPTION
**Pull request by d-x-s**
Changelog:
- **take package-lock.json from main branch**
- **voice control**
- Guided workout components
- New dependecies (carousel and MUI)
- Navbar styling compatability and bug fixes
- Remove emoji in toasts
- Refactor styles to theme file
- Animated number dial (Spring)

### CODE REVIEW

**Comment by jakji93** [comment]
image
Maybe there is a latest update that wasn't pushed, but set remaining can go less than 0 unless it is intended

File: client/src/components/Workout/GuidedExercise.jsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+              BACK TO HOME
+            </Button>
+          </Paper>
+        ) : (
+          <Box>
+            <Button
+              variant="outlined"
+              sx={styles.workoutSelectButton}
+            >
+              RESET REST
```
Comment (jakji93) [inline client/src/components/Workout/GuidedExercise.jsx:226]: reset doesn't do anything on click yet

File: client/src/components/Workout/GuidedExercise.jsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+              onClick={handleTimerClick}
+              sx={styles.workoutSelectButton}
+            >
+              {pause ? `Rest for ${restTimer} seconds` : `${restTimer} s`}
+            </Button>
+            <Button
+              variant="outlined"
+              sx={styles.workoutSelectButton}
+            >
+              +15 SECONDS
```
Comment (jakji93) [inline client/src/components/Workout/GuidedExercise.jsx:239]: same as +15, doesn't do anything right now

**Comment by d-x-s** [comment]
UI/UX Improvements:

- Finishing a set will reset the timer and start it if it is not already started
- Fixed hover styles for timer button
- Now a clear color for timer ON and timer OFF
- Fixed exercise table header colors
- Timer immediately starts counting down after click

**Review by jakji93 — APPROVED** [review]

**Review by jakji93 — DISMISSED** [review]

**Review by ckev07 — DISMISSED** [review]

**Review by jakji93 — APPROVED** [review]

**Review by jakji93 — APPROVED** [review]

**Review by kennyjhcheng — APPROVED** [review]

**Pull request merged**